src/main/java/delete_post/drivers/DeletePostDataAccess.java

```diff
@@ -1,25 +1,118 @@
 package delete_post.drivers;
 
 import delete_post.use_case.DeletePostDsGateway;
+import delete_post.use_case.PostReaderInterface;
+import entities.Post;
 
+import com.opencsv.CSVReader;
+import com.opencsv.CSVWriter;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
 import java.util.List;
 
 public class DeletePostDataAccess implements DeletePostDsGateway{
 
-    private final String path;
+    private final PostReaderInterface postReader;
+    private final String postPath;
+    private final String userPath;
 
-    public DeletePostDataAccess(String path){
-        this.path = path;
+    public DeletePostDataAccess(String postPath, String userPath, PostReaderInterface postReader){
+        this.postPath = postPath;
+        this.userPath = userPath;
+        this.postReader = postReader;
     }
 
     @Override
-    public void removeFavourite(int postId, int userId){
+    public Post getPost(int postId){
+        List<String[]> postData = readFile(postPath);
+        for (String[] row : postData) {
+            if (Integer.parseInt(row[0]) == postId) {
+                return postReader.readPost(row);
+            }
+        }
+        System.out.println("Could not find post in database");
+        return null;
+    }
 
+    public void removeFavourite(int postId, int userId){
+        removeFromUserList(postId, userId, 1);
     }
-    public void removeTag(int postId, String tag){
 
+    public void removeUser(int postId, int userId){
+        removeFromUserList(postId, userId, 2);
     }
+
     public void deletePost(int postId){
 
+        List<String[]> postData = readFile(postPath);
+
+        for (String[] row: postData){
+            if (Integer.parseInt(row[0]) == postId){
+                for (int i=1; i<9; i++){
+                    row[i] = null;
```

Comment:
Does this work when you try to write to the csv file? It seems like there may be an error when an entire row is just set to null. If it works, is there a blank line in the csv file?

Reply:
I would still suggest you copy all the posts into a new List<String[]> except the one that is to be deleted.

Reply:
the id column is still there, so its not an empty row. is that how we decided we wanted it?

Reply:
rather than setting the entire row to null, that for loop is setting columns 1-8 to null, and leaving the id column 0 as is

Reply:
I think that was how we wanted it originally but we then decided it would be better if we just delete the entire entry since we don't need it. If the id still exists, this may also be a problem for viewpost since it can't display a post where most of the entries are null. So could you please change it to delete the entire row?